From 2a44078d91667f0a45c215893830804cbadd1312 Mon Sep 17 00:00:00 2001
From: "Panwar, Ashish" <ashish.panwar@intel.com>
Date: Fri, 4 Mar 2016 11:20:00 +0530
Subject: [PATCH 1896/2508] ASoC: Intel: Skylake: Convert buffer size to # of
 u32 elements before allocating memory

commit 3bf59f7d7382a23ba83ee58fb618dfc09e5f5892 from
https://github.com/01org/linux-apollolake-i

While allocating memory, kfifo expects the number of elements in arguments
and not the # of bytes. Due to this, kfifo is currently allocating 4 times
the memory requested by user. Convert buffer size to the number of elements
before allocating memory.

Change-Id: Ib8938bc1645896a5b342aa7fe5602b4e415960a6
Signed-off-by: Panwar, Ashish <ashish.panwar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/8333
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/skylake/skl-pcm.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-pcm.c b/sound/soc/intel/skylake/skl-pcm.c
index 4cad4ae..2b933da 100644
--- a/sound/soc/intel/skylake/skl-pcm.c
+++ b/sound/soc/intel/skylake/skl-pcm.c
@@ -619,6 +619,7 @@ static int skl_trace_compr_set_params(struct snd_compr_stream *stream,
 	if (!skl_is_core_valid(core))
 		return -EINVAL;
 
+	size = size / sizeof(u32);
 	if (size & (size - 1)) {
 		dev_err(sst->dev, "Buffer size must be a power of 2\n");
 		return -EINVAL;
-- 
1.7.5.4

